scripts/bert/finetune_classifier.py

```diff
@@ -89,89 +142,140 @@
 
 ctx = mx.cpu() if not args.gpu else mx.gpu()
 
+task = tasks[task_name]
+
 # model and loss
 dataset = 'book_corpus_wiki_en_uncased'
-bert, vocabulary = bert_12_768_12(dataset_name=dataset,
-                                  pretrained=True, ctx=ctx, use_pooler=True,
-                                  use_decoder=False, use_classifier=False)
-model = BERTClassifier(bert, dropout=0.1)
-model.classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
-model.hybridize(static_alloc=True)
+bert, vocabulary = bert_12_768_12(
+    dataset_name=dataset,
+    pretrained=True,
+    ctx=ctx,
+    use_pooler=True,
+    use_decoder=False,
+    use_classifier=False)
 
-loss_function = gluon.loss.SoftmaxCELoss()
+if task.task_name in ['STS-B']:
+    model = BERTRegression(bert, dropout=0.1)
+    model.regression.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+    loss_function = gluon.loss.L2Loss()
+else:
+    model = BERTClassifier(
+        bert, dropout=0.1, num_classes=len(task.get_labels()))
+    model.classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+    loss_function = gluon.loss.SoftmaxCELoss()
+
+logging.info(model)
+model.hybridize(static_alloc=True)
 loss_function.hybridize(static_alloc=True)
-metric = mx.metric.Accuracy()
 
 # data processing
 do_lower_case = 'uncased' in dataset
 bert_tokenizer = BERTTokenizer(vocabulary, lower=do_lower_case)
 
 
-def preprocess_data(tokenizer, batch_size, dev_batch_size, max_len):
+def preprocess_data(tokenizer, task, batch_size, dev_batch_size, max_len):
     """Data preparation function."""
     # transformation
-    train_trans = BERTDatasetTransform(tokenizer, max_len,
-                                       labels=MRPCDataset.get_labels(),
-                                       pad=False, label_dtype='int32')
-    dev_trans = BERTDatasetTransform(tokenizer, max_len,
-                                     labels=MRPCDataset.get_labels(),
-                                     label_dtype='int32')
-    data_train = MRPCDataset('train').transform(train_trans, lazy=False)
-    data_dev = MRPCDataset('dev').transform(dev_trans, lazy=False)
-    data_train_len = data_train.transform(lambda input_id, length, segment_id, label_id: length)
+    train_trans = BERTDatasetTransform(
+        tokenizer,
+        max_len,
+        labels=task.get_labels(),
+        pad=False,
+        pair=task.is_pair,
+        label_dtype='float32' if not task.get_labels() else 'int32')
+    dev_trans = BERTDatasetTransform(
```

Comment:
We can reuse `train_trans ` if `dev_trans` is exactly the same as `train_trans`.

Reply:
Right!